recipes/cuda-cudart/meta.yaml

```diff
@@ -0,0 +1,108 @@
+{% set name = "cuda-cudart" %}
+{% set version = "12.0.107" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/cuda_cudart/{{ platform }}/cuda_cudart-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: c0e16e9593a69afe4997ae96a8b024e751ebce0254523fd969b08bb028582d23  # [linux64]
+  sha256: d785fd613e35e3761fdb0016a31a56417f570605cbfb9a93ca9dc68be6789188  # [ppc64le]
+  sha256: 5de248c6c203f6c91ce34256bbfc028bc87c1b0c552f9f51edb65e60ad665f63  # [aarch64]
+  sha256: 1cd8516feee068c6d718453e76b0dfcbee27844a35d4f3aa608a3d316190309c  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+  run_exports:                # [linux]
+    strong:                   # [linux]
+      - __glibc >=2.17        # [linux]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+  host:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+  run:
+    - sysroot_{{ target_platform }} 2.17   # [linux]
+    - __glibc >=2.17                       # [linux]
+
+outputs:
+  - name: cuda-cudart
+    test:
+      commands:
+        - test -f $PREFIX/lib/libcudart.so.{{ version }}     # [linux]
+        - if not exist %LIBRARY_BIN%\cudart64_*.dll exit 1   # [win]
+    files:
+      - lib/*.so.*                # [linux]
+      - targets/*/lib/*.so.*      # [linux]
+      - Library\bin               # [win]
+
+  - name: cuda-cudart-dev
+    run_exports:
+      strong:
+        - {{ pin_subpackage("cuda-cudart", max_pin='x') }}
```

Comment:
Why is this `strong`? How is this package supposed to be used in a downstream conda-build recipe?

Reply:
These are the CUDA runtime libraries. Somewhat analogous to `libgcc-ng`. Anything doing a CUDA build will link to this

Given this, how would you prefer that be specified?

Reply:
Suppose one option would be to move this to `cuda-nvcc_linux-64` similar to what is done with [`libgcc-ng`]( https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/85ab9f32b2868eddda3280956f5abcc8b8c7a021/recipe/meta.yaml#L31-L33 ). Though IIUC that would mean dropping `run_exports` from this recipe altogether.

Reply:
`strong` `run_exports` are for packages in `build` (not `host`) to appear in `run`. However if `cuda-cudart` appears in `build`, then CUDA will link to the `libcudart.so` for `build` architecture and not the `host` architecture which is wrong.

So, my previous question remains. How is this package supposed to be used in a downstream conda-build recipe?
Can you give a small recipe where you envision this package being used.

Reply:
Yeah here's [an example]( https://github.com/conda-forge/staged-recipes/pull/21350/files#r1084613040 ). Also see code around that change.

That said, the default behavior is to statically link to `cuda-cudart`, which probably makes sense to stick with. This is already what we do in CUDA builds in conda-forge. Though individual packages with different needs could use the shared library.

We can make this `run_exports/weak` I think.

Reply:
That's not a recipe that uses `cuda-cudart-dev` in the build phase. (rather in test phase). Please provide a recipe example that would use `cuda-cudart-dev` in build phase.

Reply:
That's probably going to be difficult to do without packaging the compilers and adding them to conda-forge.

In any event, think we are agreeing on switching this to `weak`. So maybe this is a non-issue at this point

Reply:
I don't mean a working recipe. I mean a recipe that you think how this package should be used.

Reply:
Yeah that makes sense

The challenge is I don't think recipes will use this directly. Other NVIDIA packages from issue ( https://github.com/conda-forge/staged-recipes/issues/21382 ) might

More likely the typical conda-forge package would use `cuda-nvcc_linux-64`, which is a bit further down the line. Though this depends a bit on how this fits in our infrastructure

Reply:
Also worth noting this has been switched to `weak`